unic-locale-impl/src/lib.rs

```diff
@@ -74,42 +78,58 @@ impl Locale {
         self.langid.get_language()
     }
 
-    pub fn set_language(&mut self, language: Option<&str>) -> Result<(), LocaleError> {
+    pub fn set_language<S: AsRef<[u8]>>(&mut self, language: S) -> Result<(), LocaleError> {
         self.langid
             .set_language(language)
             .map_err(std::convert::Into::into)
     }
 
+    pub fn clear_language(&mut self) {
+        self.langid.clear_language()
+    }
+
     pub fn get_script(&self) -> Option<&str> {
         self.langid.get_script()
     }
 
-    pub fn set_script(&mut self, script: Option<&str>) -> Result<(), LocaleError> {
+    pub fn set_script<S: AsRef<[u8]>>(&mut self, script: S) -> Result<(), LocaleError> {
         self.langid
             .set_script(script)
             .map_err(std::convert::Into::into)
     }
 
+    pub fn clear_script(&mut self) {
+        self.langid.clear_script()
+    }
+
     pub fn get_region(&self) -> Option<&str> {
         self.langid.get_region()
     }
 
-    pub fn set_region(&mut self, region: Option<&str>) -> Result<(), LocaleError> {
+    pub fn set_region<S: AsRef<[u8]>>(&mut self, region: S) -> Result<(), LocaleError> {
         self.langid
             .set_region(region)
             .map_err(std::convert::Into::into)
     }
 
+    pub fn clear_region(&mut self) {
+        self.langid.clear_region()
+    }
+
     pub fn get_variants(&self) -> Vec<&str> {
         self.langid.get_variants()
     }
 
-    pub fn set_variants(&mut self, variants: &[&str]) -> Result<(), LocaleError> {
+    pub fn set_variants<S: AsRef<[u8]>>(&mut self, variants: impl ExactSizeIterator<Item = S>) -> Result<(), LocaleError> {
         self.langid
```

Comment:
Another way to do this, maybe a bit more compact, could be (ab)using the fact that `?` calls `Into`, so:

```rust
Ok(self.langid.set_variants(variants)?)
```

Not sure if worth it.